docs/karma.md

```diff
@@ -0,0 +1,7 @@
+# Karma
+
+A currency that can be acquired by making a donation to Freenet, or by
```

Comment:
So this would be something akin to  "proof of stake"  then right? The only problem I see with this is that it allows  malicious users with enough resources crowding out normal users and allocating more bandwidth and computation to themselves.

I don't have a solution (at least not right now) about how to help out with this, but is worth noting. In this regard correct me if I am wrong but the original Freenet worked in a different way (paraphrasing from one article:  

> To  encourage  participation,  Freenet  does  not
require payment for inserts or impose restrictions
on the amount of data that publishers can insert.
Given finite disk space, however, the system must
sometimes decide which files to keep. It currently
prioritizes space allocation by popularity, as mea-
sured by the frequency of requests per file. Each
node orders the files in its data store by time of last
request, and when a new file arrives that cannot
fit in the space available, the node deletes the least
recently requested files until there is room.
...
File distribution is therefore determined by
two competing forces: tree growth and pruning.
The query-routing mechanism automatically cre-
ates more copies in an area of the network where
a file is requested, and the tree grows in that
direction. This improves response time and pre-
vents overloading when the popularity of a file
increases suddenly. Files that go unrequested in
another part of the network are subject to dele-
tion. As that part of the tree shrinks, space is
freed up for other files. The net effect is that the
number  and  location  of  copies  adjust  to  the
demand for each file.

Could this be incorporated somehow, along with diminishing returns from donations? I think diminishing returns (past certain arbitrary amount  yet to be determined; this could be even calculated dynamically) would help out with a single malicious actor crowding out others.

There is always the chance that this same actor would set up different 'fictitious' nodes and leverage this to the same goal, but this at least would create some friction to overcome (specially when the network is young).

Reply:
The "Karma" concept needs to be fleshed out and we might not even do it. I should probably be more careful about distinguishing "definite" features from more "bluesky" stuff like this.

By default we'll do things the same way the original Freenet did.

If we implement something like karma it would operate at a higher level in the network, more akin to the way Freenet's [Web of Trust](https://github.com/freenet/plugin-WebOfTrust) plugin uses captchas to prevent spam. The idea of using donations as a source of scarcity is mostly to replace the use of captchas - which can be circumvented easily using modern AI.